charts/opentelemetry-kube-stack/templates/collector.yaml

```diff
@@ -0,0 +1,241 @@
+{{ range $_, $collector := $.Values.collectors -}}
+{{- if $collector.enabled -}}
+{{- $collector := (mergeOverwrite (deepCopy $.Values.defaultCRConfig) $collector) }}
+{{- $merged := (dict "Chart" $.Chart "clusterRole" $.Values.clusterRole "collector" $collector "Release" $.Release "fullnameOverride" $.Values.fullnameOverride) }}
+{{- $fullname := (include "opentelemetry-collector.fullname" $merged) }}
+---
+apiVersion: opentelemetry.io/v1alpha1
+kind: OpenTelemetryCollector
+metadata:
+  name: {{ $fullname }}
+  namespace: {{ include "opentelemetry-kube-stack.namespace" $ }}
+  labels:
+    {{- include "opentelemetry-kube-stack.labels" $ | nindent 4 }}
+    {{- with $collector.labels }}
+    {{- range $key, $value := . }}
+    {{- printf "%s: %s" $key (tpl $value $ | quote) | nindent 4 }}
+    {{- end }}
+    {{- end }}
+  {{- with $collector.annotations }}
+  annotations:
+    {{- range $key, $value := . }}
+    {{- printf "%s: %s" $key (tpl $value $ | quote) | nindent 4 }}
+    {{- end }}
+  {{- end }}
+spec:
+  managementState: {{ $collector.managementState }}
+  mode: {{ $collector.mode }}
+  config: |
+    # if the user wants to add an entire scrape config file
+    {{- if $collector.scrape_configs_file }}
+    # template the scrape config file and load it from a YAML Array
+    {{- $loaded_config := (fromYamlArray (tpl ($.Files.Get $collector.scrape_configs_file) $)) }}
+    # create the structure for the prometheus receiver in the otel config
+    {{- $prom_override := (dict "receivers" (dict "prometheus" (dict "config" (dict "scrape_configs" $loaded_config))))}}
+    # If someone already has some scrape configs present
+    {{- if (dig "receivers" "prometheus" "config" "scrape_configs" false $collector.config) }}
+    # append the file to the set scrape configs
+    {{- $merged_prom_scrape_configs := (concat $collector.config.receivers.prometheus.config.scrape_configs $loaded_config) }}
+    # generate the structure again with the merged configs
+    {{- $prom_override = (dict "receivers" (dict "prometheus" (dict "config" (dict "scrape_configs" $merged_prom_scrape_configs)))) }}
+    {{- end }}
+    # merge the existing configuration with our override
+    {{- $new_config := (mergeOverwrite $collector.config $prom_override)}}
+    # write the config as yaml
+    {{- toYaml $new_config | nindent 4 }}
+    {{- else }}
+    # otherwise just template whats there
+    {{- toYaml $collector.config | nindent 4 }}
+    {{- end }}
+  {{- if $collector.replicas }}
```

Comment:
To reduce the number of if statements, lets set some default values in the `values.yaml` using either kubernetes defaults (like `hostNetwork: false`), and then require the value be set via the json schema.

Reply:
actually, the with covers most of these, i can probably get rid of the majority of the ifs